core/src/main/java/com/microsoft/applicationinsights/internal/jmx/JmxDataFetcher.java

```diff
@@ -83,45 +88,85 @@
     private static Collection<Object> fetch(MBeanServer server, Set<ObjectName> objects, String attributeName, String attributeType) throws Exception {
         ArrayList<Object> result = new ArrayList<Object>();
 
-        String attr = attributeName;
-        String[] inners = null;
-
         AttributeType innerAttributeType = AttributeType.REGULAR;
         if (COMPOSITE_ATTRIBUTE_TYPE.equals(attributeType)) {
             innerAttributeType = AttributeType.COMPOSITE;
         } else if (TABULAR_ATTRIBUTE_TYPE.equals(attributeType)) {
             innerAttributeType = AttributeType.TABULAR;
         }
 
-        if (innerAttributeType != AttributeType.REGULAR) {
-            inners = attributeName.split("\\.");
+        switch (innerAttributeType) {
+            case TABULAR:
+                fetchTabularObjects(server, objects, attributeName, result);
+                break;
+            case COMPOSITE:
+                fetchCompositeObjects(server, objects, attributeName, result);
+                break;
+            case REGULAR:
+                fetchRegularObjects(server, objects, attributeName, result);
+                break;
         }
 
+        return result;
+    }
+
+    static void fetchRegularObjects(MBeanServer server, Set<ObjectName> objects, String attributeName, ArrayList<Object> result)
```

Comment:
are these methods accessed outside of this class? Looks like no. If that's the case, please make these methods private.

Reply:
I'm writing some tests for these explicitly

Reply:
great please push those changes and let me know.